vnext/Desktop/WebSocket.h

```diff
@@ -49,51 +48,146 @@ class BaseWebSocket : public IWebSocket
+  ///
+  /// Dispatches tasks posted either by <see cref="m_stream" />
+  /// or arbitrary lambdas using <c>boost::asio::post</c>.
+  /// </summary>
+  /// <remarks>
+  /// Tasks will be run in the thread that calls this object's <c>run</c> method.
+  /// </remarks>
   boost::asio::io_context m_context;
+
   std::unique_ptr<boost::asio::executor_work_guard<boost::asio::io_context::executor_type>> m_workGuard;
   std::unique_ptr<Stream> m_stream;
+  std::function<void(Error&&)> m_errorHandler;
 
   BaseWebSocket(Url&& url);
 
   ~BaseWebSocket() override;
 
+  /// <summary>
+  /// Finalizes the connection setup to the remote endpoint.
+  /// Sets the ready state to <c>Open</c>.
+  /// </summary>
+  /// <remarks>
+  /// On callback, invokes the connect handler, if set.
+  /// Performs a pending write, if requested during the connection process.
+  /// Performs a pending ping call, if requested during the connection process.
+  /// Closes this instance, if requested during the connection process.
+  /// </remarks>
+  /// <param name="options">
+  /// Map of HTTP header fields sent by the remote endpoint.
+  /// </param>
   virtual void Handshake(const IWebSocket::Options& options);
 
 public:
-  #pragma region IWebSocket members
+  #pragma region IWebSocket
 
+  /// <summary>
+  /// <see cref="IWebSocket::Connect" />
+  /// </summary>
   void Connect(const Protocols& protocols, const Options& options) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::Ping" />
+  /// </summary>
   void Ping() override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::Send" />
+  /// </summary>
   void Send(const std::string& message) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::SendBinary" />
+  /// </summary>
   void SendBinary(const std::string& base64String) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::Close" />
+  /// </summary>
   void Close(CloseCode code, const std::string& reason) override;
 
   ReadyState GetReadyState() const override;
 
+  /// <summary>
+  /// <see cref="IWebSocket::SetOnConnect" />
+  /// </summary>
   void SetOnConnect(std::function<void()>&& handler) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::SetOnPing" />
+  /// </summary>
   void SetOnPing(std::function<void()>&& handler) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::SetOnSend" />
+  /// </summary>
   void SetOnSend(std::function<void(std::size_t)>&& handler) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::SetOnMessage" />
+  /// </summary>
   void SetOnMessage(std::function<void(std::size_t, const std::string&)>&& handler) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::SetOnClose" />
+  /// </summary>
   void SetOnClose(std::function<void(CloseCode, const std::string&)>&& handler) override;
+
+  /// <summary>
+  /// <see cref="IWebSocket::SetOnError" />
+  /// </summary>
   void SetOnError(std::function<void(Error&&)>&& handler) override;
 
-  #pragma endregion
+  #pragma endregion // IWebSocket
```

Comment:
>// IWebSocket [](start = 20, length = 13)

[nit] I think that as with the starting 'region' pragma above, the region name here can be specified without making it a comment.

Reply:
Noted!